### PR TITLE
make compact region tombstones ratio threshold configurable

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Previously when one region's total RocksDB tombstones count exceed the threshold, at the same time the total tombstones count exceed 50% of total entries count, we will trigger a manual compaction for this region. 
This pr make the tombstones ratio configurable and set the default ratio to 30%.

### CODE REVIEW

**Comment by siddontang** [comment]
please add description

File: etc/config-template.toml [hunk 1]
```diff
@@ -162,14 +162,22 @@
...
+
 # Number of regions for each time to check.
 # region-compact-check-step = 100
+
 # The minimum number of delete tombstones to trigger manual compaction.
 # region-compact-min-tombstones = 10000
+
+# The minimum percentage of delete tombstones to trigger manual compaction.
+# Should between 1 and 100.
+# region-compact-tombstones-ratio = 30
```
Comment (huachaohuang) [inline etc/config-template.toml]: I think it should be region-compact-tombstones-percent if we use 1~100.

Comment (breeswish) [reply etc/config-template.toml]: We can change to a f64 and use "0.3".

Comment (breeswish) [reply etc/config-template.toml]: We also need to document the precondition of triggering the compaction, i.e. `numer || ratio` or `number && ratio`.

File: src/raftstore/store/worker/compact.rs [hunk 2]
```diff
@@ -153,15 +165,20 @@ impl Runnable<Task> for Runner {
...
         return false;
     }
 
-    // When the number of tombstones exceed threshold and at least 50% entries
-    // are tombstones, this range need compacting.
+    // When the number of tombstones exceed threshold and ratio, this range need compacting.
     let estimate_num_del = num_entires - num_versions;
-    estimate_num_del >= tombstones_threshold && estimate_num_del * 2 >= num_versions
+    estimate_num_del >= tombstones_num_threshold
+        && estimate_num_del * 100 >= tombstones_percent_threshold * num_versions
```
Comment (breeswish) [inline src/raftstore/store/worker/compact.rs]: `num_versions` or `num_entires`?

**Comment by breeswish** [comment]
LGTM

**Pull request merged**